tutorandroid/plugin.py

```diff
@@ -44,17 +44,29 @@
         ),
     ]
 )
-tutor_hooks.Filters.IMAGES_PULL.add_item(
-    (
-        "android",
-        "{{ ANDROID_DOCKER_IMAGE }}",
-    )
+tutor_hooks.Filters.IMAGES_PULL.add_items(
+    [
+        (
+            "android",
+            "{{ ANDROID_DOCKER_IMAGE }}",
+        ),
+        (
+            "android-app",
+            "{{ ANDROID_APP_DOCKER_IMAGE }}",
+        ),
+    ]
 )
-tutor_hooks.Filters.IMAGES_PUSH.add_item(
-    (
-        "android",
-        "{{ ANDROID_DOCKER_IMAGE }}",
-    )
+tutor_hooks.Filters.IMAGES_PUSH.add_items(
```

Comment:
We are not going to be able to merge this change as-is. The reason is that the android-app Docker image contains user-specific configuration values that should not be stored in a public Docker registry, in most cases. This situation is identical for the MFE Docker image, which is not pushed to docker.io. In the case of the MFE image we ended up using a neat trick: 

https://discuss.overhang.io/t/why-is-the-mfe-image-not-pre-built/2260/6
https://github.com/overhangio/tutor-mfe/pull/44/files
https://github.com/overhangio/tutor-mfe/blob/ebb622fea4e0fa04ebf9173bbfed21dcdb897f15/tutormfe/plugin.py#L85

Basically, it's possible to push the MFE Docker image only when the user has overridden the default Docker image name. Maybe you can use the same trick here?

Reply:
Thanks Regis! I've made another change incorporating this filter.